main.tf

```diff
@@ -143,7 +159,38 @@ resource "aws_launch_template" "default" {
     }
   }
 
-  user_data = base64encode(templatefile("${path.module}/userdata.tpl", local.userdata_vars))
+  vpc_security_group_ids = concat(
+    [data.aws_eks_cluster.eks_cluster.vpc_config[0].cluster_security_group_id],
+    var.source_security_group_ids,
+    aws_security_group.remote_access_security_group.*.id
+  )
+
+  key_name = var.ec2_ssh_key
+
+  user_data = base64encode(
+    format("%s%s",
+      templatefile("${path.module}/userdata.tpl", local.userdata_vars),
+      (var.ami_id != null ? templatefile("${path.module}/userdata_ami.tpl", local.userdata_ami_vars) : "")
+    )
+  )
+}
+
+data "aws_eks_cluster" "eks_cluster" {
+  name = var.cluster_name
+}
+
+resource "aws_security_group" "remote_access_security_group" {
+  count = var.ec2_ssh_key != null && length(var.source_security_group_ids) == 0 ? 1 : 0
+  name        = "eks-${var.cluster_name}-${module.label.id}-remote-access"
+  description = "Security group for all nodes in the nodeGroup to allow SSH access"
+  vpc_id      = data.aws_eks_cluster.eks_cluster.vpc_config[0].vpc_id
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
 }
```

Comment:
What is the advantage of this over the previous dynamic remote access block in `aws_eks_node_group`?